api/src/exam-environment/utils/exam.ts

```diff
@@ -7,22 +7,33 @@ import {
   EnvGeneratedExam,
   EnvMultipleChoiceQuestion,
   EnvQuestionSet,
-  EnvQuestionSetAttempt,
-  user
+  EnvQuestionSetAttempt
 } from '@prisma/client';
 import { type Static } from '@fastify/type-provider-typebox';
 import * as schemas from '../schemas';
 
+interface CompletedChallengeId {
+  completedChallenges: {
+    id: string;
+  }[];
+}
+
 /**
  * Checks if all exam prerequisites have been met by the user.
- *
- * TODO: This will be done by getting the challenges required from the curriculum.
  */
-export function checkPrerequisites(_user: user, _prerequisites: unknown) {
-  return true;
+export function checkPrerequisites(
+  user: CompletedChallengeId,
```

Comment:
Pretty minor, but I would just pass around the array of completedChallenges. Reason being, this function doesn't need to know about the structure of `user`. We can make it (and its types) even simpler by only supplying the relevant data.